From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Wed, 20 Jan 2010 16:00:17 +0000 (+0100)
Subject: Resolves: #550126 - Piranha fails to parse load average with ruptime
X-Git-Url: http://git.fedorahosted.org/git/?p=piranha.git;a=commitdiff_plain;h=e6de7e51f478878dea721832ec23173710a0bf44

Resolves: #550126 - Piranha fails to parse load average with ruptime

Nanny expected line 'load average' but ruptime prints only load.

ruptime:
foo-01        up      3:51,     1 user,   load 0.07, 0.05, 0.01

uptime:
17:03:42 up  3:51,  3 users,  load average: 0.02, 0.03, 0.00

Patch by Ranjith Rajaram
---

diff --git a/nanny.c b/nanny.c
index c9f5bcf..0ac5388 100644
--- a/nanny.c
+++ b/nanny.c
@@ -841,7 +841,7 @@ getRemoteLoad (int flags, struct in_addr *remoteAddr, int timeout,
 		return -1.0;
 	}
 
-	start = strstr (output, "load average");
+	start = strstr (output, "load");
 
 	if (!start || !strchr (start, ':')) {
 		piranha_log (flags, (char *) "bad load average returned: %s",

